src/bindings/c/tests/ov_core_test.cpp

```diff
@@ -0,0 +1,280 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "ov_test.hpp"
+#include "test_model_repo.hpp"
+
+TEST(ov_version, api_version) {
+    ov_version_t version;
+    ov_get_openvino_version(&version);
+    auto ver = ov::get_openvino_version();
+
+    EXPECT_STREQ(version.buildNumber, ver.buildNumber);
+    ov_version_free(&version);
+}
+
+TEST(ov_util, ov_get_error_info_check) {
+    auto res = ov_get_error_info(ov_status_e::INVALID_C_PARAM);
+    auto str = "invalid c input parameters";
+    EXPECT_STREQ(res, str);
+}
+
+class ov_core : public ::testing::TestWithParam<std::string> {};
+INSTANTIATE_TEST_SUITE_P(device_name, ov_core, ::testing::Values("CPU"));
+
+TEST(ov_core, ov_core_create_with_config) {
+    ov_core_t* core = nullptr;
+    OV_ASSERT_OK(ov_core_create(plugins_xml, &core));
+    ASSERT_NE(nullptr, core);
+    ov_core_free(core);
+}
+
+TEST(ov_core, ov_core_create_with_no_config) {
+    ov_core_t* core = nullptr;
+    OV_ASSERT_OK(ov_core_create("", &core));
+    ASSERT_NE(nullptr, core);
+    ov_core_free(core);
+}
+
+TEST(ov_core, ov_core_read_model) {
+    ov_core_t* core = nullptr;
+    OV_ASSERT_OK(ov_core_create("", &core));
+    ASSERT_NE(nullptr, core);
+
+    ov_model_t* model = nullptr;
+    OV_ASSERT_OK(ov_core_read_model(core, xml, bin, &model));
+    ASSERT_NE(nullptr, model);
+
+    ov_model_free(model);
+    ov_core_free(core);
+}
+
+TEST(ov_core, ov_core_read_model_no_bin) {
+    ov_core_t* core = nullptr;
+    OV_ASSERT_OK(ov_core_create("", &core));
+    ASSERT_NE(nullptr, core);
+
+    ov_model_t* model = nullptr;
+    OV_ASSERT_OK(ov_core_read_model(core, xml, nullptr, &model));
+    ASSERT_NE(nullptr, model);
+
+    ov_model_free(model);
+    ov_core_free(core);
+}
+
+TEST(ov_core, ov_core_read_model_from_memory) {
```

Comment:
Can we add test to read model from memory without weights?

Reply:
But we need provide a new test model that doesn't have weights?

Reply:
Yes you can create for example model with single operation (you can hardcode it in the test examples: https://github.com/openvinotoolkit/openvino/tree/master/src/tests/functional/inference_engine/ngraph_reader).

Reply:
This test case will be put into the following PR.